mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -1199,9 +1239,9 @@ private static AssociationStorageStrategy getAssociationStorageStrategy(Associat
 	public void executeBatch(OperationsQueue queue) {
 		if ( !queue.isClosed() ) {
 			Operation operation = queue.poll();
-			Map<DBCollection, BatchInsertionTask> inserts = new HashMap<DBCollection, BatchInsertionTask>();
+			Map<MongoCollection<Document>, BatchInsertionTask> inserts = new HashMap<MongoCollection<Document>, BatchInsertionTask>();
 
-			List<Tuple> insertTuples = new ArrayList<Tuple>();
+			List<Tuple> insertTuples = new LinkedList<Tuple>();
```

Comment:
It's a great example of why unrelated changes should be sent in a different pull request:

it's not true that a LinkedList would be more suited in this case, or at least that's my opinion so - if I were to review this PR - I would start debating about this change.
Such a debate would delay the integration of a pull request which might otherwise contain nice changes. In extreme cases, having many such "distractions" would get to endless delays, and possibly conflicts which get harder to resolve when things aren't merged quickly. It's possible that large PRs with many off-topic changes never make it to be integrated..